Eastern-Research-Group/expert-query
# Feature/127 section organization

## Related Issues:
* [EQ-127](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-127)
* [EQ-142](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-142)

## Main Changes:
* Added accordion sections to the user interface.
* Re-positioned the "Clear search" button directly beneath the filter inputs.
* Added range inputs to the UI for columns with year and date values.
* Added range handling logic to the server.
* Added a check to prevent ETL from crashing on its first run.

## Steps To Test:
1. Navigate to the home page of the client side application, i.e. http://localhost:3000/attains.
2. Select the _Total Maximum Daily Load_ profile.
3. Confirm that there are 3 accordion sections, and that they operate as expected.
4. Navigate to http://localhost:9090/attains/data/tmdl?columns=completionDate&organizationName=Florida&completionDateLo=06-19-2017&completionDateHi=06-19-2017 and confirm that only rows with `completionDate` equal to `2017-06-19` are returned.
5. Navigate to http://localhost:9090/attains/data/tmdl?columns=completionDate&organizationName=Florida&completionDateLo=06-19-2017 and confirm that no rows with `completionDate` before `2017-06-19` are returned.
6. Navigate to http://localhost:9090/attains/data/tmdl?columns=completionDate&organizationName=Florida&completionDateHi=06-19-2017 and confirm that no rows with `completionDate` after `2017-06-19` are returned.
7. Navigate to http://localhost:9090/attains/data/tmdl?columns=completionDate&organizationName=Florida&completionDateLo=06-19-2017&completionDateLo=06-19-2012. The error `Error! Duplicate completionDateLo parameters not allowed` should be returned.

## TODO:
- [ ] Currently, the duplicate parameter check is being performed only for range inputs. Moving forward, this check will likely be "lifted up", and it should also be performed on `optionsParams`. Additionally, invalid parameters should be checked.
